### PR TITLE
Test/e2e tests for attribute propagation to overlay

### PR DESCRIPTION
**Pull request by rowa-audil**
Adds e2e tests for components that propagate an attribute to its overlay

### CODE REVIEW

File: apps/components-e2e/src/components/chart/pie-chart/pie-chart.e2e.ts [hunk 1]
```diff
@@ -49,6 +49,14 @@ test('should get the correct overlay values when hovering on the different pies'
     .expect(tooltip.textContent)
     .match(/15/)
     .hover(pieChart, { speed: 0.3, offsetX: 620, offsetY: 140 })
+    .wait(1000)
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/chart/pie-chart/pie-chart.e2e.ts]: if we reduce the speed to 0.1 for hovering I think we don"t need the wait here, or reduce the wait to 300 at least

File: apps/components-e2e/src/components/confirmation-dialog/confirmation-dialog.e2e.ts [hunk 2]
```diff
@@ -90,3 +91,17 @@ test('should should not open dialog when backdrop is enabled', async (testContro
...
+test('should propagate attribute to overlay', async (testController: TestController) => {
+  await testController
+    .expect(saveButton.count)
+    .eql(0)
+    .expect(dirtyDialog.count)
+    .eql(0)
+    .expect(successDialog.count)
+    .eql(0)
+    .click(openDialogButton)
+    .wait(40000)
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/confirmation-dialog/confirmation-dialog.e2e.ts]: this is definitive to high! please remove the wait. and reduce the speed at the click before to 0.3

File: apps/components-e2e/src/components/consumption/consumption.e2e.ts [hunk 3]
```diff
@@ -28,3 +33,10 @@ test('should show an overlay containing custom content while hovering and hide i
   await testController.hover(mouseoutArea);
   await testController.expect(await dummyContent.exists).notOk();
 });
+
+test('should propagate attribute to overlay', async (testController: TestController) => {
+  await testController
+    .hover(consumption)
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/consumption/consumption.e2e.ts]: please reduce hovering speed to 0.1

File: apps/components-e2e/src/components/context-dialog/context-dialog.e2e.ts [hunk 4]
```diff
@@ -18,37 +18,55 @@ import {
...
+    .ok();
 });
 
 test('should not execute click handlers when disabled', async (testController: TestController) => {
-  await testController.click(disableToggle);
-  await testController.click(contextDialog);
-  await testController.expect(await contextDialogPanel.exists).notOk();
+  await testController
+    .click(disableToggle, { speed: 0.5 })
+    .click(contextDialog)
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/context-dialog/context-dialog.e2e.ts]: please reduce speed

File: apps/components-e2e/src/components/context-dialog/context-dialog.e2e.ts [hunk 5]
```diff
@@ -18,37 +18,55 @@ import {
...
-  await testController.click(contextDialog);
-  await testController.expect(await getActiveElementText()).eql('Edit');
-
-  await testController.pressKey('tab');
-  await testController.expect(await getActiveElementText()).eql('Edit');
+  await testController
+    .click(contextDialog, { speed: 0.5 })
+    .expect(getActiveElementText())
+    .eql('Edit')
+    .pressKey('tab')
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/context-dialog/context-dialog.e2e.ts]: please reduce speed

File: apps/components-e2e/src/components/event-chart/event-chart.e2e.ts [hunk 6]
```diff
@@ -23,7 +23,7 @@ import {
   triggerChangeDetection,
 } from './event-chart.po';
 
-fixture('EventChart').page('http://localhost:4200/event-chart');
+fixture.only('EventChart').page('http://localhost:4200/event-chart');
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/event-chart/event-chart.e2e.ts]: remove `.only`

File: apps/components-e2e/src/components/select/select.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+import { select, overlayPane } from './select.po';
+
+fixture('Select').page('http://localhost:4200/select');
+
+test('should propagate attribute to overlay', async (testController: TestController) => {
+  await testController
+    .click(select)
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/select/select.e2e.ts]: please reduce speed

File: apps/components-e2e/src/components/select/select.module.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { RouterModule, Route } from '@angular/router';
+import { DtSelectModule } from '@dynatrace/barista-components/select';
+import {
+  DT_UI_TEST_CONFIG,
+  DT_DEFAULT_UI_TEST_CONFIG,
+} from 'components/core/src/testing';
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/select/select.module.ts]: please use correct import location from `@dynatrace/barista-components/core`

File: apps/components-e2e/src/components/tag/tag-add/tag-add.e2e.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+import { tagAdd, overlayPane } from './tag-add.po';
+
+fixture('Tag Add').page('http://localhost:4200/tag/tag-add');
+
+test('should propagate attribute to overlay', async (testController: TestController) => {
+  await testController
+    .click(tagAdd)
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag/tag-add/tag-add.e2e.ts]: please reduce speed

File: apps/components-e2e/src/components/tag/tag.module.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+import { NgModule } from '@angular/core';
+import { DtE2ETagAdd } from './tag-add/tag-add';
+import { CommonModule } from '@angular/common';
+import { RouterModule, Route } from '@angular/router';
+import {
+  DT_UI_TEST_CONFIG,
+  DT_DEFAULT_UI_TEST_CONFIG,
+} from 'components/core/src/testing';
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag/tag.module.ts]: wrong import location

File: apps/components-e2e/src/components/tag/tag.module.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+@NgModule({
+  declarations: [DtE2ETagAdd],
+  imports: [CommonModule, RouterModule.forChild(routes), DtTagModule],
+  exports: [],
+  providers: [
+    { provide: DT_UI_TEST_CONFIG, useValue: DT_DEFAULT_UI_TEST_CONFIG },
+  ],
+})
+export class DtE2ETagAddModule {}
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag/tag.module.ts]: Module name should be `DtE2ETagModule`

File: components/confirmation-dialog/src/confirmation-dialog.ts [hunk 12]
```diff
@@ -136,6 +145,11 @@ export class DtConfirmationDialog
     private _overlay: Overlay,
     private _viewContainerRef: ViewContainerRef,
     private _changeDetectorRef: ChangeDetectorRef,
+    /** @breaking-change: Optional will be removed with version 7.0.0 */
```
Comment (lukasholzer) [inline components/confirmation-dialog/src/confirmation-dialog.ts]: Please reword: 

```suggestion
    /** @breaking-change: `_elementRef` will be mandatory with version 7.0.0 */
```

File: components/context-dialog/src/context-dialog.ts [hunk 13]
```diff
@@ -200,6 +203,11 @@ export class DtContextDialog extends _DtContextDialogMixinBase
     @Attribute('tabindex') tabIndex: string,
     // tslint:disable-next-line: no-any
     @Optional() @Inject(DOCUMENT) private _document: any,
+    /** @breaking-change: Optional will be removed with version 7.0.0 */
```
Comment (lukasholzer) [inline components/context-dialog/src/context-dialog.ts]: Please reword: 

```suggestion
    /** @breaking-change: `_elementRef` will be mandatory with version 7.0.0 */
```

File: components/event-chart/src/event-chart.ts [hunk 14]
```diff
@@ -260,6 +264,11 @@ export class DtEventChart<T> implements AfterContentInit, OnInit, OnDestroy {
     // tslint:disable-next-line: no-any
     @Inject(DOCUMENT) private _document: any,
     private _platform: Platform,
+    /** @breaking-change: Optional will be removed with version 7.0.0 */
```
Comment (lukasholzer) [inline components/event-chart/src/event-chart.ts]: Please reword: 

```suggestion
    /** @breaking-change: `_elementRef` will be mandatory with version 7.0.0 */
```

File: components/filter-field/src/filter-field.spec.ts [hunk 15]
```diff
@@ -42,10 +42,11 @@ import {
   DtFilterFieldDefaultDataSourceType,
 } from '@dynatrace/barista-components/filter-field';
 import { DtIconModule } from '@dynatrace/barista-components/icon';
+
 import {
   dispatchFakeEvent,
-  createComponent,
   dispatchKeyboardEvent,
+  createComponent,
```
Comment (lukasholzer) [inline components/filter-field/src/filter-field.spec.ts]: importing without using?

File: components/tag/src/tag-add/tag-add.ts [hunk 16]
```diff
@@ -127,6 +134,11 @@ export class DtTagAdd implements AfterViewInit, OnDestroy {
   constructor(
     private _changeDetectorRef: ChangeDetectorRef,
     private _zone: NgZone,
+    /** @breaking-change: Optional will be removed with version 7.0.0 */
```
Comment (lukasholzer) [inline components/tag/src/tag-add/tag-add.ts]: Please reword: 

```suggestion
    /** @breaking-change: `_elementRef` will be mandatory with version 7.0.0 */
```

File: components/chart/src/tooltip/chart-tooltip.ts [hunk 17]
```diff
@@ -86,6 +94,11 @@ export class DtChartTooltip implements OnDestroy {
     private _overlay: Overlay,
     private _viewContainerRef: ViewContainerRef,
     private _changeDetectorRef: ChangeDetectorRef,
+    /** @breaking-change: Optional will be removed with version 7.0.0 */
```
Comment (lukasholzer) [inline components/chart/src/tooltip/chart-tooltip.ts]: Please reword: 

```suggestion
    /** @breaking-change: `_elementRef` will be mandatory with version 7.0.0 */
```

File: apps/components-e2e/src/components/confirmation-dialog/confirmation-dialog.module.ts [hunk 18]
```diff
@@ -19,6 +19,10 @@ import { NgModule } from '@angular/core';
 import { Route, RouterModule } from '@angular/router';
 import { DtConfirmationDialogModule } from '@dynatrace/barista-components/confirmation-dialog';
 import { DtE2EConfirmationDialog } from './confirmation-dialog';
+import {
+  DT_UI_TEST_CONFIG,
+  DT_DEFAULT_UI_TEST_CONFIG,
+} from 'components/core/src/testing';
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/confirmation-dialog/confirmation-dialog.module.ts]: wrong import location

File: apps/components-e2e/src/components/consumption/consumption.module.ts [hunk 19]
```diff
@@ -21,6 +21,10 @@ import { DtConsumptionModule } from '@dynatrace/barista-components/consumption';
 import { DtE2EConsumption } from './consumption';
 import { DtIconModule } from '@dynatrace/barista-components/icon';
 import { DtOverlayModule } from '@dynatrace/barista-components/overlay';
+import {
+  DT_UI_TEST_CONFIG,
+  DT_DEFAULT_UI_TEST_CONFIG,
+} from 'components/core/src/testing';
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/consumption/consumption.module.ts]: wrong import location

File: apps/components-e2e/src/components/context-dialog/context-dialog.module.ts [hunk 20]
```diff
@@ -19,13 +19,19 @@ import { NgModule } from '@angular/core';
 import { Route, RouterModule } from '@angular/router';
 import { DtE2eContextDialog } from './context-dialog';
 import { DtContextDialogModule } from '@dynatrace/barista-components/context-dialog';
+import {
+  DT_UI_TEST_CONFIG,
+  DT_DEFAULT_UI_TEST_CONFIG,
+} from 'components/core/src/testing';
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/context-dialog/context-dialog.module.ts]: wrong import location

File: apps/components-e2e/src/components/event-chart/event-chart.module.ts [hunk 21]
```diff
@@ -19,13 +19,19 @@ import { NgModule } from '@angular/core';
 import { Route, RouterModule } from '@angular/router';
 import { DtEventChartModule } from '@dynatrace/barista-components/event-chart';
 import { DtE2EEventChart } from './event-chart';
+import {
+  DT_UI_TEST_CONFIG,
+  DT_DEFAULT_UI_TEST_CONFIG,
+} from 'components/core/src/testing';
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/event-chart/event-chart.module.ts]: wrong import location

**Comment by rowa-audil** [comment]
Blocked by #552

**Comment by lukasholzer** [comment]
Please resolve conflicts

**Comment by tomheller** [comment]
@lukasholzer Resolved conflicts, please review again.

File: apps/components-e2e/src/components/tag/tag-add/tag-add.e2e.ts [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { tagAdd, overlayPane } from './tag-add.po';
+import { resetWindowSizeToDefault, waitForAngular } from '../../../utils';
+
+fixture
+  .only('Tag Add')
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag/tag-add/tag-add.e2e.ts]: please remove `only`

**Comment by tomheller** [comment]
Pull request has been upgraded to match angular 9. Please review @ffriedl89 @lukasholzer @thomaspink

**Comment by tomheller** [comment]
Cherry picker picked into 5.x but failed for 5.2.x because of conflicts. This is fine as these are only e2e tests for a feature that is only in 5.x.

**Pull request merged**